lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -318,7 +320,10 @@ class DefaultThumbnailsEngine(
         if (stubs.firstOrNull()?.request?.sourceId() == source && positionUs == stubs.firstOrNull()?.positionUs) {
             return
         }
-        val locatedTimestampUs = SingleThumbnailRequest(positionUs).locate(timer.durationUs.video)[0]
+        val index = dataSources[TrackType.VIDEO].indexOfFirst { it.mediaId() == source }
```

Comment:
Can we abstract it out?
May be an extension method in DataSources or directly within DataSources itself

Reply:
We've used this in multiple places. Should help simply it.